### PR TITLE
add entailment-checker

### PR DESCRIPTION
**Pull request by anakin87**
Hello @TuanaCelik,
I'm adding Fact checking rocks as you suggested...

It is a web app that incorporates a custom node, so the description may be modified/improved.

Feel free to do so!

### CODE REVIEW

**Comment by TuanaCelik** [comment]
Hey @anakin87 - if you want to, instead of this you could create a repo with just the 'EntailmentChecker' and then add this page as instructions on how to install and use the entailment checker, with a link to the fact checking rocks demo as an example

**Comment by anakin87** [comment]
Fixes #21.

I created a separate repo and package and updated this PR...

File: integrations/entailment-checker.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+## How it works
+![Entailment Checker Node](https://github.com/anakin87/haystack-entailment-checker/blob/main/images/entailment_checker_node.png)
+- The node takes a list of Documents (commonly returned by a [Retriever](https://docs.haystack.deepset.ai/docs/retriever)) and a statement as input.
+- Using a Natural Language Inference model, the text entailment between each text passage/Document (premise) and the statement (hypothesis) is computed. For every text passage, we get 3 scores (summing to 1): entailment, contradiction and neutral.
+- The text entailment scores are aggregated using a weighted average. The weight is the relevance score of each passage returned by the Retriever, if availaible. It expresses the similarity between the text passage and the statement. **Now we have a summary score, so it is possible to tell if the passages confirm, are neutral or disprove the user statement.**
+- *empirical consideration: if in the first N passages (N<K), there is strong evidence of entailment/contradiction (partial aggregate scores > **threshold**), it is better not to consider (K-N) less relevant documents.*
```
Comment (bilgeyucel) [inline integrations/entailment-checker.md:21]:
```suggestion
- *Empirical consideration: if in the first N documents (N<K), there is strong evidence of entailment/contradiction (partial aggregate scores > **threshold**), it is better not to consider the less relevant other (K-N) documents.*
```

**Review by bilgeyucel — APPROVED** [review]
Thank you @anakin87 I have one comment

**Review by bilgeyucel — APPROVED** [review]

**Pull request merged**